### PR TITLE
feat: handle old remote drives

### PR DESCRIPTION
**Pull request by gpuente**
## Description:

- Handle old remote drives
- Refactor default drives handling into DefaultDrivesManager

### CODE REVIEW

File: src/server/index.ts
Comment (acaldas) [inline src/server/index.ts:400]: Always use `logger.error` instead of `console.error`.

File: src/server/types.ts
Comment (acaldas) [inline src/server/types.ts:197]: Very nice!

File: src/server/types.ts
Comment (acaldas) [inline src/server/types.ts:233]: Let's add the driveInfo to the RemoteDriveOptions, we could call it expectedDriveInfo?

File: src/server/types.ts
Comment (acaldas) [inline src/server/types.ts:395]: With the mixins approach, the defaultDrivesManager would have access to protected methods like emit, so making it public is not needed.
Otherwise, I'd prefer that the emit method to be passed to the DefaultDrivesManager constructor, to avoid making `emit` public.
This is how the QueueManager and Storage are implemented, they receive a delegate with the methods they need to call from DocumentDriveServer.

I think that the mixins approach will work better, to prevent blocking this PR, let's use this delegate pattern. When ReadMode is merged `DefaultDrivesManager` is a good candidate for the next one to be refactored into a mixin.

File: src/storage/browser.ts
Comment (acaldas) [inline src/storage/browser.ts:116]: Other idea I explored in the ReadMode implementation, was to include the errors in the return type of the methods. Since Typescript doesn't have a way to declare what errors a method might throw, it's easy for errors to  only get caught at the root error handling, like in an ErrorBoundary in React. This makes the errors lose context.

In this case, it would return `Promise<Drive | DriveNotFoundError>` instead of throwing errors.
This way, the caller of these method is forced to handle the error case, making it easier to do error handling at the right context level.

File: src/utils/default-drives-manager.ts
Comment (acaldas) [inline src/utils/default-drives-manager.ts:30]: Maybe we could already add support for env variables at the document-drive level?
In Connect we would set `DOCUMENT_DRIVE_OLD_REMOTE_DRIVES_STRATEGY` and so on.

File: src/utils/default-drives-manager.ts
Comment (acaldas) [inline src/utils/default-drives-manager.ts:44]: `logger.error` instead.

File: src/utils/default-drives-manager.ts
Comment (acaldas) [inline src/utils/default-drives-manager.ts:175]: This would become `this.delegate?.emit` with the delegate pattern

**Comment by gpuente** [comment]
Hello @acaldas!
I think mixins is great fit for this. Let me do a refactor and address the other comments. I'll let you know if I have any question. Tks!

**Comment by gpuente** [comment]
Hey @acaldas! added the fixes here: [7724433](https://github.com/powerhouse-inc/document-drive/pull/284/commits/7724433a6235e51f7dca893aeb32b7ff5474f98b)

**Comment by gpuente** [comment]
Hey @acaldas ! I've added the fixes here: [b5ddb3c](https://github.com/powerhouse-inc/document-drive/pull/284/commits/b5ddb3cdbeb9e4ea656c6596828a2c68972366a8)

**Review by acaldas — APPROVED** [review]
Nice!

**Pull request merged**
